src/libponyc/pass/docgen.c

```diff
@@ -548,6 +548,89 @@ static void doc_type_params(docgen_t* docgen, ast_t* t_params,
     fprintf(docgen->type_file, "]");
 }
 
+// Write the expression to the current type file.
+void doc_expression(docgen_t* docgen, ast_t* ast)
+{
+  FILE* fp = docgen->type_file;
+  ast_t* child = ast_child(ast);
+
+  switch(ast_id(ast))
+  {
+    case TK_CALL:
+      while(child != NULL)
+      {
+        doc_expression(docgen, child);
+        child = ast_sibling(child);
```

Comment:
you are missing a NULL check (see later comment about duplication and ease of introducing bugs).